src/postgres/commands/copyConnectionString.ts

```diff
@@ -0,0 +1,28 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { ClientConfig } from 'pg';
+import * as vscode from 'vscode';
+import { IActionContext } from 'vscode-azureextensionui';
+import { ext } from '../../extensionVariables';
+import { localize } from '../../utils/localize';
+import { nonNullProp } from '../../utils/nonNull';
+import { PostgresDatabaseTreeItem } from '../tree/PostgresDatabaseTreeItem';
+import { checkAuthentication } from './checkAuthentication';
+
+export async function copyConnectionString(context: IActionContext, node: PostgresDatabaseTreeItem): Promise<void> {
+    if (!node) {
+        node = <PostgresDatabaseTreeItem>await ext.tree.showTreeItemPicker(PostgresDatabaseTreeItem.contextValue, context);
+    }
+
+    const clientConfig: ClientConfig = await checkAuthentication(context, node);
+
+    const credentials = nonNullProp(clientConfig, 'user') + ':' + nonNullProp(clientConfig, 'password');
+    const serverDetails = nonNullProp(clientConfig, 'host') + ':' + String(nonNullProp(clientConfig, 'port'));
+    const connectionString = "postgres://" + credentials + '@' + serverDetails + '/\'' + node.databaseName + '\'';
```

Comment:
I prefer string template literals instead of adding a bunch of string together. Feels more readable to me. What do you think of this:

```typescript
const user: string = nonNullProp(clientConfig, 'user');
const password: string = nonNullProp(clientConfig, 'user');
const host: string = nonNullProp(clientConfig, 'host');
const port: string = String(nonNullProp(clientConfig, 'port'));
const connectionString = `postgres://${user}:${password}@${host}:${port}/'${node.databaseName}'`;
```

Reply:
I agree, this is definitely better